core/templates/pages/oppia-root/app-error-handler.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Error handler for frontend.
+ */
+
+// eslint-disable-next-line oppia/disallow-httpclient
+import { HttpClient } from '@angular/common/http';
+import { ErrorHandler } from '@angular/core';
+import { LoggerService } from 'services/contextual/logger.service';
+import firebase from 'firebase/app';
+import { AppConstants } from 'app.constants';
+
+export class AppErrorHandler extends ErrorHandler {
+  private readonly UNHANDLED_REJECTION_STATUS_CODE_REGEX = (
+    /Possibly unhandled rejection: {.*"status":-1/);
+
+  MIN_TIME_BETWEEN_ERRORS_MSEC = 5000;
+  timeOfLastPostedError: number = (
+    Date.now() - this.MIN_TIME_BETWEEN_ERRORS_MSEC);
+
+  constructor(
+    private http: HttpClient,
+    private loggerService: LoggerService
+  ) {
+    super();
+  }
+
+  handleError(error: Error): void {
+    // Suppress unhandled rejection errors status code -1, because -1 is the
+    // status code for aborted requests.
+    if (this.UNHANDLED_REJECTION_STATUS_CODE_REGEX.test(
+      error.message)) {
+      return;
+    }
+
+    let errorType = Object.prototype.toString.call(error);
+    if (errorType !== '[object Error]' &&
+      errorType !== '[object DOMException]' && !(errorType instanceof Error)) {
+      // The Error.stack property provides a meaningful stacktrace of the
+      // exception. Different browsers set this value at different times.
+      // Modern browsers such as Chrome, Firefox, Edge set this value when
+      // an Error object is created. Older browsers like IE 10 & 11 set this
+      // value only when the Error is thrown. To ensure that the stack
+      // property is populated we use try/catch.
+      // eslint-disable-next-line max-len
+      // see: https://web.archive.org/web/20140210004225/http://msdn.microsoft.com/en-us/library/windows/apps/hh699850.aspx
+      try {
+        throw new Error(`${error}`);
+      } catch (errorInstance) {
+        error = errorInstance;
+      }
+    }
+
+    if (!AppConstants.DEV_MODE) {
```

Comment:
Just to check -- do you know if dropping this "if" clause would cause any issues?

Reply:
Done